src/test/java/uk/gov/companieshouse/api/accounts/controller/IntangibleAssetsControllerTest.java

```diff
@@ -0,0 +1,319 @@
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class IntangibleAssetsControllerTest {
+
+    @Mock
+    private IntangibleAssetsService intangibleAssetsService;
+
+    @Mock
+    private ApiResponseMapper apiResponseMapper;
+
+    @Mock
+    private ErrorMapper errorMapper;
+
+    @Mock
+    private IntangibleAssets intangibleAssets;
+
+    @Mock
+    private BindingResult bindingResult;
+
+    @Mock
+    private HttpServletRequest request;
+
+    @Mock
+    private Transaction transaction;
+
+    @Mock
+    private Errors errors;
+
+    @Mock
+    private SmallFull smallFull;
+
+    @Mock
+    private Map<String, String> links;
+
+    @InjectMocks
+    private IntangibleAssetsController controller;
+
+    private static final String COMPANY_ACCOUNTS_ID = "companyAccountsId";
+    private static final String INTANGIBLE_ASSETS_LINK = "intangibleAssetsLink";
+
+    @Test
+    @DisplayName("Tests the successful creation of intangible assets resource")
+    void createIntangibleAssetsSuccess() throws DataException {
+        when(bindingResult.hasErrors()).thenReturn(false);
+        when(request.getAttribute(AttributeName.TRANSACTION.getValue())).thenReturn(transaction);
+
+        ResponseObject responseObject = new ResponseObject(ResponseStatus.CREATED, intangibleAssets);
+        when(intangibleAssetsService.create(intangibleAssets, transaction, COMPANY_ACCOUNTS_ID, request))
+                .thenReturn(responseObject);
+
+        ResponseEntity responseEntity = ResponseEntity.status(HttpStatus.CREATED).body(responseObject.getData());
+        when(apiResponseMapper.map(responseObject.getStatus(), responseObject.getData(), responseObject.getErrors()))
+                .thenReturn(responseEntity);
+
+        ResponseEntity response =
+                controller.create(intangibleAssets, bindingResult, COMPANY_ACCOUNTS_ID, request);
+
+        assertNotNull(response);
+        assertEquals(HttpStatus.CREATED, response.getStatusCode());
+        assertEquals(intangibleAssets, response.getBody());
+
+        verify(errorMapper, never()).mapBindingResultErrorsToErrorModel(bindingResult);
+    }
+
+    @Test
+    @DisplayName("Tests the creation of a intangible assets resource when binding result errors are present")
+    void createIntangibleAssetsBindingResultErrors() throws DataException {
+
+        when(bindingResult.hasErrors()).thenReturn(true);
+        when(errorMapper.mapBindingResultErrorsToErrorModel(bindingResult)).thenReturn(errors);
+
+        ResponseEntity response =
+                controller.create(intangibleAssets, bindingResult, COMPANY_ACCOUNTS_ID, request);
+
+        assertNotNull(response);
+        assertEquals(HttpStatus.BAD_REQUEST, response.getStatusCode());
+        assertEquals(errors, response.getBody());
+
+        verify(errorMapper, times(1)).mapBindingResultErrorsToErrorModel(bindingResult);
```

Comment:
Again, you've mocked this behaviour so no need to verify this function has been called

Reply:
Sorted